twisted/web/test/test_http2.py

```diff
@@ -1971,6 +1972,30 @@ def validate(streamID):
         return a._streamCleanupCallbacks[1].addCallback(validate)
 
 
+    def test_unnecessaryWindowUpdateForStream(self):
+        """
+        When a WindowUpdate frame is received for a stream but no data is
+        currently waiting, nothing exciting happens.
+        """
+        f = FrameFactory()
+        transport = StringTransport()
+        conn = H2Connection()
+        conn.requestFactory = DummyHTTPHandler
+
+        # Send the request.
+        frames = buildRequestFrames(
+            self.postRequestHeaders, self.postRequestData, f
+        )
+        frames.insert(1, f.buildWindowUpdateFrame(streamID=1, increment=5))
```

Comment:
I guess that we can introduce the new DSL helpers in this ticket and if they work out well we can roll them to other tests.

For me is important to have tests which are simple and easy to read as we don't have any automated code to test the test code. The only way to check that a test is valid is by reading the code.

Here we create 2 frames, then we insert a new frame between the newly created 2 frames and then we send only the first 2 frames.

So the code would be

```
frames = []
frames.append(f.buildHeadersFrame(streamID=1, self.postRequestHeaders))
frames.append(f.buildWindowUpdateFrame(streamID=1, increment=5))

conn.dataReceived(serializeFrames(frames))
```